### PR TITLE
contrib/docker: cleanup wrapper a bit

### PR DESCRIPTION
**Pull request by rubenk**
- No need to declare errno
- Remove one level of indentation

### CODE REVIEW

**Comment by rubenk** [comment]
@mfournier what do you think?

File: contrib/docker/rootfs_prefix/rootfs_prefix.c [hunk 1]
```diff
@@ -11,28 +11,22 @@
...
-      return orig;
-    } else {
-      return (const char *)prefixed;
-    }
+  if ((strncmp(orig, "/proc", 5) != 0) && (strncmp(orig, "/sys", 4) != 0))
+    return orig;
 
-  } else {
+  int status = snprintf(prefixed, BUFSIZE, "%s%s", PREFIX, orig);
+  if ((unsigned int)status >= BUFSIZE) {
```
Comment (octo) [inline contrib/docker/rootfs_prefix/rootfs_prefix.c]: You should do the `status < 1` check first: status may be negative due to error. When cast to an *unsigned int*, the resulting value is likely larger than `BUFSIZE`, resulting in an incorrect error message.

File: contrib/docker/rootfs_prefix/rootfs_prefix.c [hunk 2]
```diff
@@ -11,28 +11,22 @@
...
-      return orig;
-    } else if (status < 1) {
-      error(status, errno,
-            "adding '%s' prefix to file path failed: '%s' -> '%s'", PREFIX,
-            orig, prefixed);
-      return orig;
-    } else {
-      return (const char *)prefixed;
-    }
+  if ((strncmp(orig, "/proc", 5) != 0) && (strncmp(orig, "/sys", 4) != 0))
```
Comment (octo) [inline contrib/docker/rootfs_prefix/rootfs_prefix.c]: I find `strlen("/proc")` easier to read than `5`.

File: contrib/docker/rootfs_prefix/rootfs_prefix.c (hunk 1 above)
Comment (rubenk) [inline contrib/docker/rootfs_prefix/rootfs_prefix.c]: Ah duh, thanks for spotting this. Wil fix.

**Comment by cwolfinger** [comment]
Doesnt this leak memory on each time the file is open ?

**Pull request merged**